src/win/fs.c

```diff
@@ -764,8 +764,87 @@ void fs__mkdir(uv_fs_t* req) {
 }
 
 
-/* OpenBSD original: lib/libc/stdio/mktemp.c */
+static void fs__mkdtemp_callback(uv_fs_t* req, static const size_t num_x, WCHAR* ep) {
+  size_t len;
+
+  if (_wmkdir(req->file.pathw) == 0) {
+    len = strlen(req->path);
+    wcstombs((char*) req->path + len - num_x, ep - num_x, num_x);
+    SET_REQ_RESULT(req, 0);
+    return 1;
+  } else if (errno != EEXIST) {
+    SET_REQ_RESULT(req, -1);
+    return 1;
+  }
+
+  return 0;
+}
+
+
 void fs__mkdtemp(uv_fs_t* req) {
+  fs__mktemp(req, fs__mkdtemp_callback);
+}
+
+
+int fs__mkstemp_callback(uv_fs_t* req, static const size_t num_x, WCHAR* ep) {
+  size_t len;
+  HANDLE file;
+  int fd;
+
+  file = CreateFileW(req->file.pathw,
+                     NULL,
+                     NULL,
```

Comment:
The second and third parameter of CreateFile are DWORDs not pointers (NULL).
And you should pass meaningful values for dwShareMode and dwDesiredAccess. `dwShareMode` should probably be `FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE` in order to be consistent with the other functions. dwDesiredAccess `GENERIC_READ | GENERIC_WRITE` or something similar.